LucyWilcoxGrocreyList/app/src/main/java/com/example/lwilcox/lucywilcoxgrocerylist/MainActivity.java

```diff
@@ -0,0 +1,102 @@
+package com.example.lwilcox.lucywilcoxgrocerylist;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+
+    private Button saveButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        ListView listView = (ListView) findViewById(R.id.listview);
+
+        final ArrayList<String> grocreyList = new ArrayList<String>();
+
+        saveButton = (Button) findViewById(R.id.button);
+
+        final ArrayAdapter<String> itemsAdapter =
```

Comment:
Instead of declaring final variables in the middle of your Activity Class, you can make these Class Member variables by defining `ArrayAdapter<String> itemsAdapter;` outside of these methods (underneath your `saveButton`)